### ISSUE TITLE
Tests are failing

### ISSUE THREAD

**Issue by laurentlb** [bazelbuild/rules_webtesting#352]
According to Bazel CI, rules_webstesting doesn't work with Bazel flag `--incompatible_depset_is_not_iterable`:

https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/110#f5e4c0cc-36b5-44b2-80fe-9b3ead98c239

Can you take a look?

**Comment by laurentlb** [comment]
I cannot reproduce locally. Our CI ran the tests at commit 24771d41071fbd7483e5c81160f3f400cbfa06e4, which is old. This probably happens because tests are failing.

**Comment by laurentlb** [comment]
```
$ bazel build --nobuild ...
Starting local Bazel server and connecting to it...
ERROR: /.../laurentlb/rules_webtesting/go/metadata/BUILD.bazel:21:1: in go_library rule //go/metadata:go_default_library: non-test target '//go/metadata:go_default_library' depends on testonly target '//go/bazel:go_default_library' and doesn't have testonly attribute set
ERROR: /.../laurentlb/rules_webtesting/go/metadata/BUILD.bazel:21:1: in go_library rule //go/metadata:go_default_library: non-test target '//go/metadata:go_default_library' depends on testonly target '//go/metadata/capabilities:go_default_library' and doesn't have testonly attribute set
ERROR: Analysis of target '//testdata:https' failed; build aborted: Analysis of target '//go/metadata:go_default_library' failed; build aborted
```

**Comment by hlopko** [comment]
Hi @joshbruning, thanks for these fixes! It seems it didn't help though :( https://buildkite.com/bazel/rules-webtesting-saucelabs/builds/119#7453627d-1941-4eb3-88ee-0d696ac14f00

**Comment by DrMarcII** [comment]
Fixed as of PR #358
